File: Try_GameEngine5/src/com/example/try_gameengine/remotecontroller/UpKeyCommandPressDown.java
package com.example.try_gameengine.remotecontroller;

import android.view.MotionEvent;

import com.example.try_gameengine.remotecontroller.RemoteController.CommandType;

public class UpKeyCommandPressDown implements Command{
	UpKey upKey;
	private int pointerId = -1;
	
	public UpKeyCommandPressDown(UpKey upKey){
		this.upKey = upKey;
	}
	
	@Override
	public CommandType execute() {
		return CommandType.UPKeyDownCommand;
	}

	@Override
	public boolean checkExecute(float x, float y, MotionEvent event) {
		return upKey.pressUp(x, y, event);
	}

	@Override
	public void setMotionEventPointerId(int pointerId) {
		this.pointerId = pointerId;
	}

	@Override
	public int getMotionEventPointerId() {
		return pointerId;
	}

}

File: Try_GameEngine5/src/com/example/try_gameengine/remotecontroller/UpKeyCommandPressDown.kt
package com.example.try_gameengine.remotecontroller

import android.view.MotionEvent

class UpKeyCommandPressDown(upKey: UpKey) : Command {
    var upKey: UpKey
    private var pointerId = -1

    init {
        this.upKey = upKey
    }

    override fun execute(): RemoteController.CommandType {
        return RemoteController.CommandType.UPKeyDownCommand
    }

    override fun checkExecute(x: Float, y: Float, event: MotionEvent?): Boolean {
        return upKey.pressUp(x, y, event)
    }

    override fun setMotionEventPointerId(pointerId: Int) {
        this.pointerId = pointerId
    }

    override fun getMotionEventPointerId(): Int {
        return pointerId
    }
}
